src/validator/LookupValidator.cpp

```diff
@@ -397,6 +402,24 @@ StatusOr<Value> LookupValidator::checkConstExpr(Expression* expr, const std::str
     auto type = schema->getFieldType(prop);
     QueryExpressionContext dummy(nullptr);
     auto v = Expression::eval(expr, dummy);
+    // Allow different numeric type to compare
+    if (graph::SchemaUtil::propTypeToValueType(type) == Value::Type::FLOAT && v.isInt()) {
+        return v.toFloat().first;
+    } else if (graph::SchemaUtil::propTypeToValueType(type) == Value::Type::INT && v.isFloat()) {
+        // col1 < 10.5 range: [min, 11), col1 < 10 range: [min, 10)
+        // cast float to int and add 1 to get the ceilling
+        if ((leftIsAE && (kind == Expression::Kind::kRelGE || kind == Expression::Kind::kRelLT)) ||
+            (!leftIsAE && (kind == Expression::Kind::kRelGT || kind == Expression::Kind::kRelLE))) {
+            auto castVal = v.toInt().first;
+            // edge case col1 >= 40.0, no need to round up
+            if (castVal == v.getFloat()) {
+                return castVal;
+            }
+            return castVal + 1;
+        }
+        return v.toInt().first;
+    }
+
```

Comment:
A small suggestion, It's better to extract a method, such as : 
Value castToTargetType(const Value& v, PropertyType targetType) {
swith (v.getType()) {
case int, float {
 return cast v from current type to target type if needed.
}
case ….. {
return bad value;
}
}
}

Reply:
The reason I didn't extract a function is that the only case we allow different types to compare is when both are numerical, specifically, int and float. I'll modify this part if we decide to support more cross-type comparisons. 
Does it make sense to you?